pkg/controller/azvolume.go

```diff
@@ -230,9 +230,8 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 		}
 	}()
 
-	// only try deleting underlying volume 1) if volume creation was successful and 2) volumeDeleteRequestAnnotation is present
-	// if the annotation is not present, only delete the CRI and not the underlying volume
-	if isCreated(azVolume) && mode == detachAndDeleteCRI {
+	// delete CRI only when either 1) volume is not created, 2) driver is uninstalling, or 3) volume is pre-provisioned
```

Comment:
It seems to no longer be the case that we delete the CRI if the volume is pre-provisioned.